src/bundle/ui-dev/src/modules/universal-discovery/components/tree-item-toggle-selection/tree.item.toggle.selection.js

```diff
@@ -28,12 +28,14 @@ const TreeItemToggleSelection = ({ locationId, isContainer, contentTypeIdentifie
     }
 
     const [selectedLocations, dispatchSelectedLocationsAction] = useContext(SelectedLocationsContext);
-    const [multiple] = useContext(MultipleConfigContext);
+    const [multiple, multipleItemsLimit] = useContext(MultipleConfigContext);
     const containersOnly = useContext(ContainersOnlyContext);
     const allowedContentTypes = useContext(AllowedContentTypesContext);
     const restInfo = useContext(RestInfoContext);
+    const isSelected = selectedLocations.some((selectedLocation) => selectedLocation.location.id === locationId);
     const isNotSelectable =
         (containersOnly && !isContainer) || (allowedContentTypes && !allowedContentTypes.includes(contentTypeIdentifier));
+    const isSelectionBlocked = multipleItemsLimit !== 0 && !isSelected && selectedLocations.length >= multipleItemsLimit;
```

Comment:
you've created helper for it (`src/bundle/ui-dev/src/modules/universal-discovery/helpers/selected.locations.helper.js`) couldn't you use it here, or difference is structure?

Reply:
These items are not necessarily locations so I didn't want to use location-specific helpers.

Reply:
Helpers from `selected.locations.helper.js` use `location` but in this component we have only access to location's id, which would be sufficient for `selected.locations.helper.js`, but these helpers were designed with location parameter and I don't want to change it now, as this can result in BC.